legend-engine-pure/legend-engine-pure-ide/legend-engine-pure-ide-light-http-server/src/main/java/org/finos/legend/engine/ide/PureIDEServer.java

```diff
@@ -155,5 +161,50 @@ public PureSession getPureSession()
         return pureSession;
     }
 
-    protected abstract MutableList<RepositoryCodeStorage> buildRepositories(SourceLocationConfiguration sourceLocationConfiguration);
+    protected MutableList<RepositoryCodeStorage> buildRepositories(SourceLocationConfiguration sourceLocationConfiguration)
+    {
+        Map<CodeRepository, Path> codeRepositoryPathMap = new HashMap<>();
+        List<String> directoriesToSearch = sourceLocationConfiguration.directories != null && !sourceLocationConfiguration.directories.isEmpty()
+                                            ? sourceLocationConfiguration.directories
+                                            : Lists.mutable.of(".");
+
+        for (String path : directoriesToSearch)
+        {
+            Map<CodeRepository, Path> p = findCodeRepositoriesAndMapToPath(path);
+            codeRepositoryPathMap.putAll(p);
+        }
+
+        MutableList<RepositoryCodeStorage> result = Lists.mutable.empty();
+        for (CodeRepository r : codeRepositoryPathMap.keySet())
+        {
+            if (r instanceof GenericCodeRepository) //TODO is this correct/needed?
+            {
+                Path repoDefFilePath = codeRepositoryPathMap.get(r);
+                String fileName = repoDefFilePath.getFileName().toString();
+                String dirName = fileName.substring(0, fileName.indexOf('.'));
+                Path parentDirectory = repoDefFilePath.getParent();
+
+                result.add(new MutableFSCodeStorage(r, parentDirectory.resolve(dirName)));
+            }
+        }
+
+        return result;
+    }
+
+    //TODO: This should probably be moved to CodeRepositoryProviderHelper in legend-pure
+    protected Map<CodeRepository, Path> findCodeRepositoriesAndMapToPath(String path)
+    {
+        Map<CodeRepository, Path> result = new HashMap<>();
+        try
+        {
+            Path pth = Paths.get(path);
+            List<Path> defFiles = Files.walk(pth).filter(p -> !p.toString().contains("archetype") && !p.toString().contains("target") && p.toString().endsWith(".definition.json")).collect(Collectors.toList());
+            defFiles.forEach(p -> result.put(GenericCodeRepository.build(p), p));
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
```

Comment:
Actually, something like this might be better, since you can more precisely exclude directory trees.
```suggestion
        Map<CodeRepository, Path> result = Maps.mutable.empty();
        Set<String> skipDirs = Sets.mutable.with("target", "archetype");
        try
        {
            Files.walkFileTree(Paths.get(path), new SimpleFileVisitor<Path>()
            {
                @Override
                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributs attrs)
                {
                    return skipDirs.contains(dir.getFileName().toString()) ? FileVisitResult.SKIP_SUBTREE : FileVisitResult.CONTINUE;
                }
                
                @Override
                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
                {
                    if (file.toString().endsWith(".definition.json"))
                    {
                        result.put(GenericCodeRepository.build(file), file);
                    }
                    return FileVisitResult.CONTINUE;
                }
            });
        }
        catch (IOException e)
        {
            throw new UncheckedIOException(e);
        }
```

Reply:
Better yet, actually, would be to allow users to configure excluded paths with globs in the config file, and then use FileSystem.getPathMatcher("glob:" + excl) to create a glob based PathMatcher.

Reply:
Suggestion added in subsequent commit - took the glob pattern approach.